### PR TITLE
Arch arm assembly simplifications

### PR DESCRIPTION
**Pull request by ioannisg**
Preliminaries for #19689

### CODE REVIEW

File: arch/arm/core/swap_helper.S [hunk 1]
```diff
@@ -427,7 +427,7 @@ SECTION_FUNC(TEXT, z_arm_svc)
 
     ldr r1, [r0, #24]   /* grab address of PC from stack frame */
     /* SVC is a two-byte instruction, point to it and read encoding */
```
Comment (agansari) [inline arch/arm/core/swap_helper.S]: change this comment to reference reading the lower byte of SVC

**Review by agansari — CHANGES_REQUESTED** [review]

File: arch/arm/core/swap_helper.S (hunk 1 above)
Comment (ioannisg) [reply arch/arm/core/swap_helper.S]: Right, thanks. Fixed now

**Review by ioannisg — COMMENTED** [review]
@agansari pls, take a look once more, and thanks for reviewing

**Review by agansari — APPROVED** [review]

**Review by wentongwu — APPROVED** [review]

**Review by tyrantlizard77 — APPROVED** [review]
cool!

**Pull request merged**